src/app/gateways/views/GatewayTrayView.vue

```diff
@@ -0,0 +1,80 @@
+<template>
```

Comment:
This is not something to do with this PR as it existed before but I'd like to understand.

Why do we have different lists, details and trays for Gateway and Regular dataplanes? Aren't they the same in the end? At least from a backend POV there are gateway DPPs and regular DPPs.

Looks like clicking on a detail leads me to:
https://deploy-preview-1646--kuma-gui.netlify.app/gui/meshes/port-2/gateways/firewall-185eca577b-cf1ae.card-gateway-0/overview

it feels like it should lead me to:
https://deploy-preview-1646--kuma-gui.netlify.app/gui/meshes/port-2/data-planes/firewall-185eca577b-cf1ae.card-gateway-0/overview

We may render things differently depending on the type of DPP but I don't think these things should be fundamentally different

Reply:
They’re very similar, yes, but in the past we had quite a few bugs related to the kinds of conditionals that differentiate between regular DPPs and gateway DPPs. That is still the case in some places. Most notably the DataPlaneList.vue component.

Now I don’t have super strong opinions on whether we do "share the components for regular and gateway DPPs" or "try and keep the separate components to avoid conditionals", so we could change this (in a different PR preferably).

One more thing that might influence that is whether we turn Gateways into their separate resource in which case changing this makes less sense. But of course that might just be too far into the future.

@johncowen Also here, any opinions? I know you were somewhat keen on making this split also?

Reply:
I'm not sure if I follow @lahabana s question entirely, but..

If I follow what @kleinfreund is saying, I think in the case of the DataPlaneList component, it at least used to contain a fair amount of code to do something different depending on whether something was a gateway or not. At the time we had discussed using the same DataPlaneList component for yet another use, so dataplanes, gateways _and_ dataplanes plus gateways. So I suggested not having a reusable dataplane list component at all (moving the reused stuff to the data layer), and just using our Table component in the View for all three. That would also have helped with prevent any issues (both technical and UX) around this area.

If it turns out that these aren't fundamentally different, then maybe we just reuse the route, or maybe a component in different routes or maybe just slightly different usages of out table component in each route. Is it worth even having a gateway tab? Maybe we can just filter a list of DPPs by gateway.

I think first we should figure out what we want to see, which I think is what the related issue is about.

@kleinfreund is the recent bugfix related this in this PR preview?

Reply:
I’ve been talking with @lahabana about this again.

To summarize, there are some tentative (non-concrete) plans around this that involve two things:

- We should have one listing (i.e. the Data Plane Proxies tab) that shows all resources for its associated endpoint (i.e. including what we currently split-off into the Gateways tab. They are the same resource and only have minor differences (and importantly: much more in common).
- We should have a listing (i.e. the Gateways tab) that shows MeshGateway resources instead (caveat to solve: those would only be for one of the gateway types; I forget whether delegated or provided)

In other words, definitely not something to do anything about here.

Reply:
I don't have problems with different lists as indeed the content is quite different.
What I'm more concerned about is Tray and Details because there's just more commonality in these views than there are differences.

I think the more I'm thinking about it the more "list gateway dataplane" is likely to move away. In favour of just "list gateway services". 

Also there are discussions about having "gateway" just a property of an inbound so a dataplane proxy could be both a gateway and a regular dataplane...